src/stateManager/StateManager.ts

```diff
@@ -482,11 +484,34 @@ class StateManager {
         if (!this.agreementManager.didEveryoneSignBlock(block)) {
             console.log("Posting calldata on chain!");
             this.p2pEventHooks.onPostingCalldata?.();
-            this.stateChannelManagerContract
-                .postBlockCalldata(signedBlock, Clock.getTimeInSeconds())
-                .then((txResponse) => txResponse.wait())
+
+            await call(() =>
+                this.stateChannelManagerContract.postBlockCalldata(
+                    signedBlock,
+                    Clock.getTimeInSeconds()
```

Comment:
`maxTimestamp` that's passed to `postBlockCalldata` is `previousBlock.getRelevantTimestamp() + p2pTime + agreementTime + chainFallbacktime`